erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -78,6 +78,15 @@ frappe.ui.form.on('Payment Entry', {
 				filters: { "name": ["in", doctypes] }
 			};
 		});
+
+		frm.set_query("reference_name", "references", function () {
+			return {
+				filters: {
+					docstatus: 1,
+					title: frm.doc.party
```

Comment:
@ashwinisave35,

filter document based on `party type` instead of `title`. e.g. if Party type is Customer then filter based on customer